internal/config/config.go

```diff
@@ -935,9 +936,48 @@ func validateBGPAdvPerPool(adv *BGPAdvertisement, pool *Pool) error {
 				"this pool is more specific than the aggregation length for addresses %s", adv.AggregationLength, lowest, addr)
 		}
 	}
+
+	// Verify that BGP ADVs set a unique local preference value per BGP update.
+	for _, bgpAdv := range pool.BGPAdvertisements {
+		if adv.LocalPref != bgpAdv.LocalPref {
+			if adv.AggregationLength == bgpAdv.AggregationLength || adv.AggregationLengthV6 == bgpAdv.AggregationLengthV6 {
+				if !uniqueConfig(adv, bgpAdv) {
+					return fmt.Errorf("invalid local preference %d: local preferernce %d was "+
+						"already set for the same type of BGP update. Check existing BGP advertisements "+
+						"with common pools and aggregation lengths", adv.LocalPref, bgpAdv.LocalPref)
+				}
+			}
+		}
+	}
+
 	return nil
 }
 
+func uniqueConfig(newAdv, adv *BGPAdvertisement) bool {
```

Comment:
let's call this "advertisementsAreCompatible"

Reply:
done